src/CodeFormatter.cs

```diff
@@ -187,18 +201,20 @@ private static void LogWorkspaceDiagnostics(ILogger logger, bool logWorkspaceWar
             }
         }
 
-        private static async Task<Solution> RunCodeFormattersAsync(
+        private static async Task<(Solution Solution, List<FormattedFile> FormattedFiles)> RunCodeFormattersAsync(
             Solution solution,
             ImmutableArray<(DocumentId, OptionSet, ICodingConventionsSnapshot)> formattableDocuments,
             FormatOptions options,
             ILogger logger,
             CancellationToken cancellationToken)
         {
-            var formattedSolution = solution;
+            var formattedSolution = (Solution: solution, FormattedFiles: new List<FormattedFile>());
 
             foreach (var codeFormatter in s_codeFormatters)
             {
-                formattedSolution = await codeFormatter.FormatAsync(formattedSolution, formattableDocuments, options, logger, cancellationToken).ConfigureAwait(false);
+                var formatterResponse = await codeFormatter.FormatAsync(formattedSolution.Solution, formattableDocuments, options, logger, cancellationToken).ConfigureAwait(false);
+                formattedSolution.Solution = formatterResponse.Solution;
+                formattedSolution.FormattedFiles.AddRange(formatterResponse.FormattedFiles);
```

Comment:
Why not just give the code formatter a list and have it add its formatted files to it instead of returning them in a tuple?